app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx

```diff
@@ -129,53 +124,71 @@ export function Dialog(props: types.DialogProps) {
-                    </aria.Header>
-                  )}
-
-                  <div className="relative flex-auto overflow-y-auto p-3.5">
-                    {typeof children === 'function' ? children(opts) : children}
-                  </div>
-                </>
-              )}
-            </aria.Dialog>
+              <aria.Dialog
+                id={dialogId}
+                ref={mergeRefs.mergeRefs(dialogRef, element => {
+                  if (element) {
+                    // This is a workaround for the `data-testid` attribute not being
+                    // supported by the 'react-aria-components' library.
+                    // We need to set the `data-testid` attribute on the dialog element
+                    // so that we can use it in our tests.
+                    // This is a temporary solution until we refactor the Dialog component
+                    // to use `useDialog` hook from the 'react-aria-components' library.
+                    // this will allow us to set the `data-testid` attribute on the dialog
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                    element.dataset.testId = testId
+                  }
+                })}
+                className={dialogSlots.base()}
+                {...ariaDialogProps}
+              >
+                {opts => (
+                  <dialogProvider.DialogProvider value={{ close: opts.close, dialogId }}>
+                    {shouldRenderTitle && (
+                      <aria.Header className={dialogSlots.header()}>
+                        <ariaComponents.CloseButton
+                          className={clsx('col-start-1 col-end-1 mr-auto mt-0.5', {
+                            hidden: hideCloseButton,
+                          })}
+                          onPress={opts.close}
+                        />
+
+                        <aria.Heading
+                          slot="title"
+                          level={2}
+                          className="col-start-2 col-end-2 my-0 text-base font-semibold leading-6"
+                        >
+                          {title}
+                        </aria.Heading>
+                      </aria.Header>
+                    )}
+
+                    <div className="relative flex-auto overflow-y-auto p-3.5">
+                      <errorBoundary.ErrorBoundary>
+                        <React.Suspense fallback={<loader.Loader minHeight="h32" />}>
+                          {typeof children === 'function' ? children(opts) : children}
+                        </React.Suspense>
+                      </errorBoundary.ErrorBoundary>
+                    </div>
+                  </dialogProvider.DialogProvider>
+                )}
+              </aria.Dialog>
+            </dialogStackProvider.DialogStackRegistrar>
           </aria.Modal>
         )
       }}
     </aria.ModalOverlay>
   )
 }
+
+const TYPE_TO_DIALOG_TYPE: Record<
```

Comment:
constants belong at the very top of the file

Reply:
I'd love to keep it below just because this constant is not that necessary to stay on top, imo